### PR TITLE
QPPA-921: Add QPP header and footer to Developer Documentation

### PR DESCRIPTION
**Pull request by samskeller**
Adding the basic QPP header and footer to the developer documentation, with slight differences (which means we couldn't use the Header and Footer that come with the qpp-style repo) -- the header does't have the MIPS/APMs/About dropdowns and the footer has links, which are relative links in the qpp-style repo.

The rest looks basically the same -- see below for some screenshots. The only thing that's not quite working is getting the "Subscribe To Updates" modal to actually show when you click on the link. The modal works just fine once it's up, it's just not being triggered to go up when you click the link in the footer. I believe this is because I'm currently leaving out Boostrap's JS file, which I can't seem to include because I get errors about jQuery not being present (even though it's imported before Bootstrap). Once this is solved, I think this should be good to go.

Very open to hearing suggestions on reorganization as I've never really worked on a React app before!

I'm also not sure how to properly include the `qpp-style.css` file, since it's generated from within the node modules... should we pull it out and include in our own repo?

Header screenshots:
1:
screen shot 2017-08-14 at 9 57 49 pm

2:
screen shot 2017-08-14 at 9 59 56 pm

Footer screenshots:
1:
screen shot 2017-08-14 at 9 58 14 pm

2:
screen shot 2017-08-14 at 9 58 34 pm

3:
screen shot 2017-08-14 at 9 58 23 pm

### CODE REVIEW

File: src/styles/app.css [hunk 1]
```diff
@@ -36,3 +50,25 @@ h2 a.tutorial-header-link {
...
+.qpp-docs-title-container {
+	background: #036;
+	padding-top: 30px;
+	padding-bottom: 10px;
+}
+
+.qpp-docs-title-content-container {
+	margin-left: auto;
+	margin-right: auto;
+	max-width: 1100px;
```
Comment (abarciauskas-bgse) [inline src/styles/app.css]: Can we dry this up with `.container` above - in other words remove the duplicate attributes use the `.container` class above `<div className="qpp-docs-title-content-container container">`

File: src/styles/app.css [hunk 2]
```diff
@@ -13,12 +22,17 @@ table td {
...
-.container {
+.content-container {
   margin-left: auto;
   margin-right: auto;
   max-width: 1100px;
   width: 90%;
 }
+.container {
+  margin-left: auto;
+  margin-right: auto;
```
Comment (abarciauskas-bgse) [inline src/styles/app.css]: Could be more terse and combine these lines into:
```
margin: 0px auto;
```

File: src/styles/app.css [hunk 3]
```diff
@@ -36,3 +50,25 @@ h2 a.tutorial-header-link {
 h2:hover a.tutorial-header-link {
   text-decoration: underline;
 }
+
+.qpp-docs-title-container {
+	background: #036;
```
Comment (abarciauskas-bgse) [inline src/styles/app.css]: Nit but here and below there are 4 spaces of indentation and above we use 2 spaces.

File: src/components/footer.jsx [hunk 4]
```diff
@@ -3,31 +3,44 @@ import React from 'react';
...
+            <div className="col-sm-8 col-md-9 col-lg-9">
+              <div className="logo-container">
+                <img className="qpp-logo" src="https://qpp.cms.gov/images/qpp_logo_reversed.png" alt="qpp logo"></img>
               </div>
-              <div className="usa-footer-primary-content usa-footer-contact_info">
-                <a href="mailto:info@agency.gov">info@agency.gov</a>
+              <div className="links">
+                <ul>
+                  <li>
+                    <a href="https://qpp.cms.gov/developers" data-track-category="FooterNav" data-track-action="OpenDeveloperTools" data-track-label="Quality Payment Program">Developer Tools</a>
```
Comment (kencheeto) [inline src/components/footer.jsx]: what do `data-track-category` and `data-track-action` do?

File: src/components/footer.jsx [hunk 5]
```diff
@@ -3,31 +3,44 @@ import React from 'react';
...
+                  </li>
+                  <li>
+                    <a href="https://qpp.cms.gov/about/resource-library" data-track-category="FooterNav" data-track-action="OpenEducationAndTools" data-track-label="Education and Tools">Resource Library</a>
+                  </li>
+                  <li>
+                    <a href="mailto:QPP@cms.hhs.gov">Send US Questions</a>
```
Comment (kencheeto) [inline src/components/footer.jsx]: Send United States Questions? Send Us Questions? 🤔

File: src/components/app.jsx [hunk 6]
```diff
@@ -1,13 +1,20 @@
...
 import url from 'url';
+
+import '../../node_modules/jquery/src/jquery';
 import '../../node_modules/uswds/dist/js/uswds.js';
+import '../../node_modules/bootstrap/dist/js/bootstrap.js';
+
 import '../../node_modules/@cmsgov/design-system-core/dist/index.css';
+import '../../node_modules/bootstrap/dist/css/bootstrap.css';
 
+import '../styles/qpp-style/css/qpp-style.css';
```
Comment (kencheeto) [inline src/components/app.jsx]: could we import these in the header and footer files, or does app.jsx use qpp styles?

File: src/components/app.jsx [hunk 7]
```diff
@@ -1,13 +1,20 @@
 import React from 'react';
 import url from 'url';
+
+import '../../node_modules/jquery/src/jquery';
 import '../../node_modules/uswds/dist/js/uswds.js';
+import '../../node_modules/bootstrap/dist/js/bootstrap.js';
+
 import '../../node_modules/@cmsgov/design-system-core/dist/index.css';
+import '../../node_modules/bootstrap/dist/css/bootstrap.css';
```
Comment (samskeller) [inline src/components/app.jsx]: Good question! Doesn't look like it -- removed them in 07ff302961da611d50249f4c276b9c3ce99c2281

File: src/components/app.jsx (hunk 6 above)
Comment (samskeller) [inline src/components/app.jsx]: Yeah I do actually use it in app.jsx, so have to import it here

File: src/styles/app.css (hunk 3 above)
Comment (samskeller) [inline src/styles/app.css]: Yeah seems like a good idea! I don't know much about sass but I know that a lot of people use it

File: src/components/footer.jsx (hunk 4 above)
Comment (samskeller) [inline src/components/footer.jsx]: Good question! As far as I can tell... nothing. I was just copying them from the qpp-style header/footer, but that's a bad reason to use them. Removed them in 07ff302961da611d50249f4c276b9c3ce99c2281!

File: src/components/footer.jsx (hunk 5 above)
Comment (samskeller) [inline src/components/footer.jsx]: lololol typo -- fixed in 07ff302961da611d50249f4c276b9c3ce99c2281

**Pull request closed without merging**